core/src/main/java/greencity/service/impl/TipsAndTricksServiceImpl.java

```diff
@@ -0,0 +1,167 @@
+
+    private final FileService fileService;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AddTipsAndTricksDtoResponse save(AddTipsAndTricksDtoRequest addTipsAndTricksDtoRequest, MultipartFile image,
+                                            String email) {
+        TipsAndTricks toSave = modelMapper.map(addTipsAndTricksDtoRequest, TipsAndTricks.class);
+        toSave.setAuthor(userService.findByEmail(email));
+        if (addTipsAndTricksDtoRequest.getImage() != null) {
+            image = convertToMultipartImage(addTipsAndTricksDtoRequest.getImage());
+        }
+        if (image != null) {
+            toSave.setImagePath(fileService.upload(image).toString());
+        }
+        toSave.setTipsAndTricksTags(addTipsAndTricksDtoRequest.getTipsAndTricksTags()
+            .stream()
+            .map(tipsAndTricksTagsService::findByName)
+            .collect(Collectors.toList())
+        );
+
+        try {
+            tipsAndTricksRepo.save(toSave);
+        } catch (DataIntegrityViolationException e) {
+            throw new NotSavedException(ErrorMessage.TIPS_AND_TRICKS_NOT_SAVED);
+        }
+
+        return modelMapper.map(toSave, AddTipsAndTricksDtoResponse.class);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PageableDto<TipsAndTricksDto> findAll(Pageable page) {
+        Page<TipsAndTricks> pages = tipsAndTricksRepo.findAllByOrderByCreationDateDesc(page);
+        List<TipsAndTricksDto> tipsAndTricksDtos = pages
+            .stream()
+            .map(tipsAndTricks -> modelMapper.map(tipsAndTricks, TipsAndTricksDto.class))
+            .collect(Collectors.toList());
+
+        return new PageableDto<>(
+            tipsAndTricksDtos,
+            pages.getTotalElements(),
+            pages.getPageable().getPageNumber()
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PageableDto<TipsAndTricksDto> find(Pageable page, List<String> tags) {
+        Page<TipsAndTricks> pages = tipsAndTricksRepo.find(page, tags);
+
+        List<TipsAndTricksDto> tipsAndTricksDtos = pages.stream()
+            .map(tipsAndTricks -> modelMapper.map(tipsAndTricks, TipsAndTricksDto.class))
+            .collect(Collectors.toList());
+
+        return new PageableDto<>(
+            tipsAndTricksDtos,
+            pages.getTotalElements(),
+            pages.getPageable().getPageNumber()
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TipsAndTricks findById(Long id) {
+        return tipsAndTricksRepo
+            .findById(id)
+            .orElseThrow(() -> new NotFoundException(ErrorMessage.TIPS_AND_TRICKS_NOT_FOUND_BY_ID + id));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TipsAndTricksDto findDtoById(Long id) {
+        TipsAndTricks tipsAndTricks = findById(id);
+        return modelMapper.map(tipsAndTricks, TipsAndTricksDto.class);
```

Comment:
I don't see why do you need these 2 methods - as for me first one is enough.

Reply:
Replaced findById by other existing methods not to duplicate code